add -check tc flag; dynamically find sac2c SO

hv15 opened the pull request:
With this PR we add `-check tc` flag which activates some runtime checks on array bounds and indices.

Additionally, we revamp how sac2c is detected and loaded (current mechanism work only for global installs of sac2c).

Should resolve #5.

EDIT: we also resolve #4.

hv15 committed:
add -check tc flag; dynamically find sac2c SO

hv15 commented:
@ashinkarov using `-check tc` with AUD seems to be faulty (https://github.com/SacBase/sac-jupyter/issues/5#issuecomment-1075048850). How bad is the performance with AKD specialization vs. AUD?

ashinkarov commented:
@hv15, one could give it a go, but as far as I remember, the CNN tutorial was getting embarrassingly slow to do without aud (and mildly slow with aud).  The right thing of course would be to fix `-checktc`, but this might take a while.  Maybe for now, just extend the text of `%help` to indicate that checks can be turned on with these flags, but leave the defaults as is?

sbliven commented:
I have the first two lessons from the main tutorial in my [examples](https://github.com/sbliven/sac-jupyter/tree/examples) branch if this helps with testing. AKD mode took 53s to run 86 cells, AUD only 35s.

I agree that the error message for `[1][1]` is opaque. However isn't it more confusing if they run this and get -233994656?

hv15 committed:
change to -check c (with aud specmode)

As per comment https://github.com/SacBase/sac-jupyter/issues/5#issuecomment-1076103155

hv15 commented:
@ashinkarov as per https://github.com/SacBase/sac-jupyter/issues/5#issuecomment-1076103155 I've changed specmode to AUD, and instead only do `-check c`. I will as part of the new jupyter docker image re-compile sac stdlib with `-check c`, which should make specmode AKD unnecessary.

hv15 committed:
refactor

sbscholz reviewed (approved):
looks all fine; now we need to make sure that we use a stdlib which is compiled with `-check c`.

ashinkarov commented:
Looks good, except the `LD_LIBRARY_PATH` thingy, not sure whether there is a good fix.  Have a look, and let me know when you are done, I am happy to merge this in.
